aiida/orm/groups.py

```diff
@@ -318,100 +348,39 @@ def get_or_create(cls, backend=None, **kwargs):
 
         return cls.objects(backend).get_or_create(**kwargs)
 
-    @classmethod
-    def query(cls,
-              name=None,
-              type_string="",
-              id=None,
-              uuid=None,
-              nodes=None,
-              user=None,
-              node_attributes=None,
-              past_days=None,
-              backend=None,
-              **kwargs):  # pylint: disable=too-many-arguments, redefined-builtin, invalid-name
-        """
-        Query for groups.
-
-        :note: By default, query for user-defined groups only (type_string=="").
-            If you want to query for all type of groups, pass type_string=None.
-            If you want to query for a specific type of groups, pass a specific
-            string as the type_string argument.
-
-        :param name: the name of the group
-        :param nodes: a node or list of nodes that belongs to the group (alternatively,
-            you can also pass a DbNode or list of DbNodes)
-        :param id: the pk of the group
-        :param uuid: the uuid of the group
-        :param type_string: the string for the type of node; by default, look
-            only for user-defined groups (see note above).
-        :param user: by default, query for groups of all users; if specified,
-            must be a DbUser object, or a string for the user email.
-        :param past_days: by default, query for all groups; if specified, query
-            the groups created in the last past_days. Must be a datetime object.
-        :param node_attributes: if not None, must be a dictionary with
-            format {k: v}. It will filter and return only groups where there
-            is at least a node with an attribute with key=k and value=v.
-            Different keys of the dictionary are joined with AND (that is, the
-            group should satisfy all requirements.
-            v can be a base data type (str, bool, int, float, ...)
-            If it is a list or iterable, that the condition is checked so that
-            there should be at least a node in the group with key=k and
-            value=each of the values of the iterable.
-        :param backend: the backend to query
-        :param kwargs: any other filter to be passed to DbGroup.objects.filter
-
-            Example: if ``node_attributes = {'elements': ['Ba', 'Ti'], 'md5sum': 'xxx'}``,
-                it will find groups that contain the node with md5sum = 'xxx', and moreover
-                contain at least one node for element 'Ba' and one node for element 'Ti'.
-
-        """
-        return cls.objects(backend).group_query(
-            name=name,
-            type_string=type_string,
-            id=id,
-            uuid=uuid,
-            nodes=nodes,
-            user=user,
-            node_attributes=node_attributes,
-            past_days=past_days,
-            **kwargs)
-
     @classmethod
     def get_from_string(cls, string):
         """
```

Comment:
In think this should be deprecated as people should use the `get` method with `label=label`

Reply:
Ok,

in which version will it be removed then (for the warning message)?

Reply:
We currently don't mention version number in deprecation messages. Not good practice, I know

Reply:
ok, deprecated